influxdb3_server/src/query_executor.rs

```diff
@@ -573,17 +585,247 @@ impl std::fmt::Debug for SystemSchemaProvider {
 }
 
 impl SystemSchemaProvider {
-    fn new(_catalog: Arc<Catalog>, _query_log: Arc<QueryLog>, include_debug_info: bool) -> Self {
-        let tables = HashMap::new();
+    fn new(_catalog: Arc<Catalog>, query_log: Arc<QueryLog>, include_debug_info: bool) -> Self {
+        let mut tables = HashMap::<&'static str, Arc<dyn TableProvider>>::new();
         if include_debug_info {
-            // Using todo!() here causes gRPC integration tests to fail, likely because they
-            // enable debug mode by default, thus entering this if block. So, just leaving this
-            // here in lieu of todo!().
-            //
-            // Eventually, we will implement the queries and parquet_files tables and they will
-            // be injected to the provider's table hashmap here...
-            info!("TODO - gather system tables");
+            // TODO - remaining system tables gathered here...
+            let queries = Arc::new(SystemTableProvider::new(Arc::new(QueriesTable::new(
+                query_log,
+            ))));
+            tables.insert(_QUERIES_TABLE, queries);
         }
         Self { tables }
     }
 }
+
+#[async_trait]
+impl SchemaProvider for SystemSchemaProvider {
+    fn as_any(&self) -> &dyn Any {
+        self as &dyn Any
+    }
+
+    fn table_names(&self) -> Vec<String> {
+        let mut names = self
+            .tables
+            .keys()
+            .map(|s| (*s).to_owned())
+            .collect::<Vec<_>>();
+        names.sort();
+        names
+    }
+
+    async fn table(&self, name: &str) -> Result<Option<Arc<dyn TableProvider>>, DataFusionError> {
+        Ok(self.tables.get(name).cloned())
+    }
+
+    fn table_exist(&self, name: &str) -> bool {
+        self.tables.contains_key(name)
+    }
+}
+
+struct QueriesTable {
+    schema: SchemaRef,
+    query_log: Arc<QueryLog>,
+}
+
+impl QueriesTable {
+    fn new(query_log: Arc<QueryLog>) -> Self {
+        Self {
+            schema: queries_schema(),
+            query_log,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl IoxSystemTable for QueriesTable {
+    fn schema(&self) -> SchemaRef {
+        Arc::clone(&self.schema)
+    }
+
+    async fn scan(&self) -> Result<RecordBatch, DataFusionError> {
+        let schema = self.schema();
+
+        let entries = self
+            .query_log
+            .entries()
+            .entries
+            .into_iter()
+            .map(|e| e.state())
+            .collect::<Vec<_>>();
+
+        from_query_log_entries(Arc::clone(&schema), &entries)
+    }
+}
+
+fn queries_schema() -> SchemaRef {
```

Comment:
Arguably, this function, as well as the following, could be moved to core to be shared between distributed/monolith.